bin/mag.js

```diff
@@ -0,0 +1,88 @@
+
+db.init(function(err) {
+    if(err) throw err;
+    rcon = redis.createClient(config.redis.port, config.redis.server);
+    rcon.on('error', err=>{throw err});
+    rcon.on('ready', ()=>{
+        logger.info("connected to redis");
+        setInterval(health_check, 1000*60*2); //start checking health 
+        setTimeout(health_check, 1000*10); //run shortly after start
+        run();
+    });
+});
+
+var report = {
+    status: "ok",
+    maxage: 1000*60*30,
+    project_counts: 0,
+}
+
+function health_check() {
+    report.date = new Date();
+    rcon.set("health.warehouse.projectinfo."+process.env.HOSTNAME+"-"+process.pid, JSON.stringify(report));
+}
+
+function run() {
+	db.Projects.find({
+        removed: false,
+    })
+    //.populate('app project')
+    .exec((err, projects)=>{
+		if(err) throw err;
+        report.project_counts = projects.length;
+        async.eachSeries(projects, handleProject, err=>{
+            if(err) logger.error(err);
+            console.log("done going through all projects sleeping.....");
```

Comment:
As we discussed,  I think we can just run this script via cron - so you don't have to persist it. Also, if we run it via cron, we don't need to do the health reporting / redis publishing because corn can route error message. You can get rid of all those mechanisms.